references/hub.py

```diff
@@ -0,0 +1,112 @@
+# See LICENSE or go to <https://www.apache.org/licenses/LICENSE-2.0.txt> for full license details.
+
+# Inspired by: https://github.com/rwightman/pytorch-image-models/blob/master/timm/models/hub.py
+
+import json
+import os
+import subprocess
+from pathlib import Path
+from typing import Any, Dict
+
+from huggingface_hub import HfApi, HfFolder, Repository
+
+from doctr.file_utils import is_tf_available, is_torch_available
+
+
+def login_to_hub() -> None:
+    access_token = HfFolder.get_token()
+    if access_token is not None and HfApi()._is_valid_token(access_token):
+        print("Huggingface Hub token found and valid")
+        HfApi().set_access_token(access_token)
+    else:
+        subprocess.call(['huggingface-cli', 'login'])
+        HfApi().set_access_token(HfFolder().get_token())
+    # check if git lfs is installed
+    try:
+        subprocess.call(['git', 'lfs', 'version'])
+    except FileNotFoundError:
+        raise OSError('Looks like you do not have git-lfs installed, please install. \
+                      You can install from https://git-lfs.github.com/. \
+                      Then run `git lfs install` (you only have to do this once).')
+
+
+def save_model_and_config_for_hf_hub(model: Any, save_dir: str, model_config: Dict[str, str]) -> None:
+    save_directory = Path(save_dir)
+
+    if is_torch_available():
+        import torch
+        weights_path = save_directory / 'pytorch_model.bin'
+        torch.save(model.state_dict(), weights_path)
+    elif is_tf_available():
+        import tensorflow as tf  # noqa: F401
+        weights_path = save_directory / 'tf_model' / 'weights'
+        model.save_weights(str(weights_path))
+    else:
+        raise RuntimeError("Neither PyTorch nor TensorFlow is available.")
+
+    config_path = save_directory / 'config.json'
+
+    with config_path.open('w') as f:
+        json.dump(model_config, f, indent=2, ensure_ascii=False)
+
+
+def push_to_hf_hub(model: Any, model_name: str, tag: str, run_config: Any) -> None:
+
+    # default readme
+    readme = f"""
+---
+language: en
+---
+
+<p align="center">
+  <img src="https://github.com/mindee/doctr/releases/download/v0.3.1/Logo_doctr.gif" width="60%">
+</p>
+
+**Optical Character Recognition made seamless & accessible to anyone, powered by TensorFlow 2 & PyTorch**
+
+## Task: {tag}
+
+https://github.com/mindee/doctr
+
+### Example usage:
+
+```python
+>>> from doctr.io import DocumentFile
+>>> from doctr.models import ocr_predictor
+>>> from doctr.models.<task> import from_hub
+>>>
+>>> img = DocumentFile.from_images(['<image_path>'])
+>>> # Load your model from the hub
+>>> model = from_hub('mindee/my-recognition-model').eval()
+>>> # Pass it to the predictor
+>>> predictor = ocr_predictor(det_arch='db_mobilenet_v3_large',
+>>>                           reco_arch=model,
+>>>                           pretrained=True)
+>>> # Get your predictions
+>>> res = predictor(img)
+```
```

Comment:
I think it is not the right docstring there

Reply:
@charlesmindee 
Yes i think we have to change it ftm to TODO or anything else until classifiaction/detection/recognition have a `from_hub`

I'm fine with a change to models/factory but i am not sure how we can test it in fact that we would need a huggingface test account for this 

Example from model card currently
![Screenshot from 2022-04-04 11-45-14](https://user-images.githubusercontent.com/23098070/161518713-d4b289d2-dc84-442a-9a3b-d32bac4e5183.png)

Reply:
@charlesmindee 

sry i think you misunderstood :) this is a readme file which will be added to the uploaded model-repository but now i have added also docstrings :+1: